control/src/nodelets/mushr_feedback_controller.cpp

```diff
@@ -0,0 +1,165 @@
+#include <ml4kp_bridge/defs.h>
+#include <ros/ros.h>
+
+#include <nodelet/nodelet.h>
+#include <pluginlib/class_list_macros.h>
+
+#include "prx_models/mj_mushr.hpp"
+#include <control/MushrControlPropagation.h>
+#include <ml4kp_bridge/TrajectoryStamped.h>
+#include <prx_models/MushrObservation.h>
+#include <std_msgs/Empty.h>
+
+namespace control
+{
+class mushr_feedback_controller_t : public nodelet::Nodelet
+{
+public:
+  mushr_feedback_controller_t()
+  {
+    std::string params_file;
+    ros::param::get("/ml4kp_params", params_file);
+    std::cout << "Loading params from: " << params_file << std::endl;
+    auto ml4kp_params = prx::param_loader(params_file);
+    auto controller_params = prx::param_loader(ml4kp_params["feedback_controller"].as<std::string>());
+
+    prx::simulation_step = ml4kp_params["simulation_step"].as<double>();
+    std::string plant_name = ml4kp_params["/plant/name"].as<>();
+    std::string plant_path = ml4kp_params["/plant/path"].as<>();
+    plant = prx::system_factory_t::create_system(plant_name, plant_path);
+    ROS_ASSERT(plant != nullptr);
+
+    std::vector<double> min_state_limits = ml4kp_params["/plant/state_space/lower_bound"].as<std::vector<double>>();
+    std::vector<double> max_state_limits = ml4kp_params["/plant/state_space/upper_bound"].as<std::vector<double>>();
+    plant->get_state_space()->set_bounds(min_state_limits, max_state_limits);
+    std::vector<double> param_values = ml4kp_params["/plant/parameter_space/values"].as<std::vector<double>>();
+    plant->get_parameter_space()->copy_from(param_values);
+
+    // Control space limits of the real MuSHR
+    std::vector<double> min_control_limits = { -1., -1. };
+    std::vector<double> max_control_limits = { 1., 1. };
+    plant->get_control_space()->set_bounds(min_control_limits, max_control_limits);
+
+    // prx::world_model_t control_model({ plant }, {});
+    std::vector<prx::system_ptr_t> systems = { plant };
+    std::vector<std::shared_ptr<prx::movable_object_t>> obstacles = {};
+    control_model = std::make_shared<prx::world_model_t>(systems, obstacles);
+    control_model->create_context("control_context", { plant_name }, {});
+    context = control_model->get_context("control_context");
+    ROS_ASSERT(context.first != nullptr);
+
+    trajectory = std::make_shared<prx::trajectory_t>(plant->get_state_space());
+    current_state = plant->get_state_space()->make_point();
+
+    if (controller_params["name"].as<std::string>() == "mushr_pure_pursuit")
+    {
+      controller = new prx::mushr_pure_pursuit_t(plant, controller_params);
+    }
+    else if (controller_params["name"].as<std::string>() == "mushr_stanley")
+    {
+      controller = new prx::mushr_stanley_t(plant, controller_params);
+    }
+    else
+    {
+      ROS_ERROR("Unknown controller type: %s", controller_params["name"].as<std::string>().c_str());
+      return;
+    }
+    controller->set_goal(ml4kp_params["goal_state"].as<std::vector<double>>());
```

Comment:
There is no templated set_goal in the main branch of ML4KP, is there a PR missing?